### ISSUE TITLE
Link to oj2534-hw?

### ISSUE THREAD

**Issue by rusefi** [fenugrec/oj2534-fw#1]
How alive is all this and where could oj2534-hw be found? :)

**Comment by fenugrec** [comment]
Hi, thanks for your interest - it's not quite dead but on long term standby.

- hw isn't complete, I have a fairly robust K interface, a bunch of designs for the J2534 voltage-mode pin driver,  but couldn't decide on the rest of the power chain -- buck+boost + linear stage, SEPIC + lin stage,  SEPIC only... that's where I stopped.
- project on standby because :
-- in ~4 years, nobody has shown an interest in contributing, only getting a free product
-- if I try to sell some of these to try and justify the amount of work, it's going to be hard to sell them - china has a flood of cheap clones for under 50$.
-- if I try to sell them, I'm going to get requests for all kinds of stuff I don't care about ( DCAN, LIN, insane Chrysler SCI garbage, J1850, the list goes on...)

I still think this is a great idea, but I can't afford to give away a few hundred hours of work for someone to copy the design and sell for cheaper than mine.

I'm open to suggestions though !

**Comment by rusefi** [comment]
@fenugrec are you making money with your other open source projects or is it just fun? with my rusefi project... well... it's only for fun - I just accept that I only do this instead of watching football.

Where are you located? Would you be available to give me a 101 on vehicle interfaces? I am kind of maybe thinking to add support into my open source ECU.

**Comment by fenugrec** [comment]
>other open source projects

They are 99% hobby, but I don't really need a J2534 interface anymore so this project would become 100% work : I have tons of other "more fun" projects I'd like to work on in my spare time.

> interfaces 101

I think it can be explained pretty simply (applied to your ECU project) - go with CAN, a.k.a iso15765 "iso-TP". K line is simple but old, slow, and painful to work with. Avoid that and the other protocols like the plague (J1850, SCI, K line...)
With iso-TP I think I've seen open software stacks already, so it might not be too hard to implement.

**Comment by rusefi** [comment]
Thank you, will look into iso15765 "iso-TP"

**Comment by fenugrec** [comment]
Good luck ! If someone from your side feels like working on a J2534 interface, send them here P-)

**Comment by yipperr** [comment]
won't  it be easier writing a firmware for the machnna m2 device or any other ones like the chipkit boards  with a pic mcu and can transceivers  to read the j2534 api calls in firmware and do the bus protocol on the board (essentially just firmware devlopment then)

Do you mind elaborating what your project entails ? don't you need a mcu that has a can controller with flash to hold the firmware and the can tranceivers to implement it from scratch , but in the above description you don't seem to mention anything about can , does the hardware you envision not have can ? 

sorry if this is the wrong place to ask these questions and thank you for your time

**Comment by fenugrec** [comment]
> won't it be easier writing a firmware for the machnna m2 device or any other ones
> [...] (essentially just firmware devlopment then)

yes and no. it can be done, yes the M2 device has most of the hardware necessary (it didn't exist / I hadn't heard about it when I started this project). *however* the M2 and other simple devices don't have the programmable voltage pin drivers that some J2534 vehicles require.

> don't you need a mcu that has a can controller with flash to hold the firmware and the can tranceivers

The stm32F072 device I am using for tests has built-in CAN and USB peripherals. Add a cheap TJA1050 / MCP2562 transceiver. Could do the same for K line (mc33290 etc) but those ICs are generally harder to find and not cheap, so I designed a fairly robust discrete transceiver with a dual comparator and a handful of jellybean components.

Other than that, my plan was to have the mcu firmware handle the shitty iso9141 / 14230 timing requirements +  do a small amount of hardware CAN filtering + handle periodic messages; the rest would be in the host driver that implements the j2534 API.
